dev/com.ibm.ws.security.ready.service/src/com/ibm/ws/security/ready/SecurityReadyService.java

```diff
@@ -10,16 +10,27 @@
  *******************************************************************************/
 package com.ibm.ws.security.ready;
 
+import java.util.concurrent.CountDownLatch;
+
 /**
  *
  */
 public interface SecurityReadyService {
     /**
      * Answers if the security service as a whole is ready to process requests.
-     * 
+     *
      * @return boolean indiciating if the security service is ready to process
      *         requests.
      */
     public boolean isSecurityReady();
 
+    /**
+     * Provides a CountDownLatch that can be used to wait for the
+     * security service as a whole to be ready
+     *
+     * @return CountDownLatch that countdowns when security service is ready to
+     *         process requests
+     */
+    public CountDownLatch getSecurityReadyCDL();
```

Comment:
Instead of `getSecurityReadCDL()`, could we add a method `await(long timeout, TimeUnit unit)`, the the caller doesn't need to be aware of the CDL implementation detail, and all callers would not need to duplicate the logic to perform the await operation.